.github/workflows/e2e.yml

```diff
@@ -13,30 +13,61 @@ jobs:
         scenario-filter: ["^TestDappDevelop$/^.*$"]
     runs-on: ubuntu-latest
     env:
-      TOOLS_REF: ${{ github.ref }}
-      SYSTEM_TEST_GIT_REF: v1.0.6
-      # core git ref is latest commit for stable soroban functionality
-      SYSTEM_TEST_CORE_GIT_REF: 871accefc0c4a703c1321b4f0e48cf6e03b41960
-      SYSTEM_TEST_CORE_COMPILE_CONFIGURE_FLAGS: "--disable-tests --enable-next-protocol-version-unsafe-for-production"
+      # the gh tag of system-test repo version to run 
+      SYSTEM_TEST_GIT_REF: v1.0.7
+
+      # the soroban tools source code to compile and run from system test
+      # refers to checked out source of current git hub ref context 
+      SYSTEM_TEST_SOROBAN_TOOLS_REF: ${{ github.workspace }}/soroban-tools
+
+      # core git ref should be latest commit for stable soroban functionality
+      # the core bin can either be compiled in-line here as part of ci, 
+      # SYSTEM_TEST_CORE_GIT_REF: 871accefc0c4a703c1321b4f0e48cf6e03b41960
+      # SYSTEM_TEST_CORE_COMPILE_CONFIGURE_FLAGS: "--disable-tests --enable-next-protocol-version-unsafe-for-production"
+      # or can use option to pull a pre-compiled image instead, namely to save time during ci run
+      SYSTEM_TEST_CORE_IMAGE: sreuland/system-test-core:871accefc0c4a
```

Comment:
@sreuland this doesn't really improve the dependency eval. We should build all from source while respecting the inter-repo dependencies.
i.e.
core <- horizon (go) <- soroban-tools ( rpc ) <- system-tests

otherwise, we'll be updating this file indefinitely.

Reply:
yes, in near term, soroban enabled core docker builds not published to registry yet, trying this out to measure the time savings during ci workflow run if image were available, will change back to use compile, when images are available later, the config can be updated here to take advantage.